tools/ping1d_mavlink_driver.py

```diff
@@ -0,0 +1,137 @@
+                    help="Minimum acceptable confidence percentage for depth measurements.\""
+                    )
+ARGS = PARSER.parse_args()
+
+def main():
+    """ Main function
+    """
+
+    ## The time that this script was started
+    tboot = time.time()
+
+    ## Parser to decode incoming PingMessage
+    ping_parser = PingMessage.PingParser()
+
+    ## Messages that have the current distance measurement in the payload
+    distance_messages = [
+        PingMessage.PING1D_DISTANCE,
+        PingMessage.PING1D_DISTANCE_SIMPLE,
+        PingMessage.PING1D_PROFILE
+        ]
+
+    ## The minimum interval time for distance updates to the autopilot
+    ping_interval_ms = 0.1
+
+    ## The last time a distance measurement was received
+    last_distance_measurement_time = 0
+
+    ## The last time a distance measurement was requested
+    last_ping_request_time = 0
+
+    pingargs = ARGS.ping.split(':')
+    pingserver = (pingargs[0], int(pingargs[1]))
+
+    autopilot_io = mavutil.mavlink_connection("udpout:" + ARGS.mavlink,
+                                              source_system=1,
+                                              source_component=192
+                                              )
+
+    ping1d_io = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
+    ping1d_io.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, 1)
+    ping1d_io.setblocking(False)
+
+    ## Send a request for distance_simple message to ping device
+    def send_ping1d_request():
+        data = PingMessage.PingMessage()
+        data.request_id = PingMessage.PING1D_DISTANCE_SIMPLE
+        data.src_device_id = 0
+        data.packMsgData()
+        ping1d_io.sendto(data.msgData, pingserver)
+
+    # some extra information for the DISTANCE_SENSOR mavlink message fields
+    min_distance = 20
+    max_distance = 5000
+    sensor_type = 2
+    orientation = 25
+    covarience = 0
+
+    ## Send distance_sensor message to autopilot
+    def send_distance_data(distance, deviceid, confidence):
+        print("sending distance %d confidence %d" % (distance, confidence))
+        if confidence < ARGS.min_confidence:
+            distance = 0
+
+        autopilot_io.mav.distance_sensor_send(
+            (time.time() - tboot) * 1000, # time_boot_ms
+            min_distance, # min_distance
+            max_distance, # max_distance
+            distance/10, # distance
+            sensor_type, # type
+            deviceid, # device id
+            orientation,
+            covarience)
+
+    while True:
+        time.sleep(0.001)
+        tnow = time.time()
+
+        # request data from ping device
+        if tnow > last_distance_measurement_time + ping_interval_ms:
+            if tnow > last_ping_request_time + ping_interval_ms:
+                last_ping_request_time = time.time()
+                send_ping1d_request()
+
+        # read data in from ping device
+        try:
+            data, _ = ping1d_io.recvfrom(4096)
```

Comment:
Btw, you can check for data on the buffer with 'select', so you only try to read when there is data ready

Reply:
Yes, and that requires an additional dependency. Is there anything wrong with what we are doing here? Is it functionally any different from what select is doing?

Reply:
Select is part of the Python standard library.
I've ran into issues by using this type of code in the past,  but sadly I can't really remember what the issue was.

I have no problem with leaving it as it is if it works, let's just keep the alternative in mind if any issues arise.

Reply:
I'm not aiming to change the code algorithm but only styles corrections. 
I can send a second just with the code corrections here.